src/libraries/Common/src/Interop/Linux/os-release/Interop.OSReleaseFile.cs

```diff
@@ -0,0 +1,82 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+
+internal static partial class Interop
+{
+    // Parse information from '/etc/os-release'.
+    internal static class OSReleaseFile
+    {
+        private const string EtcOsReleasePath = "/etc/os-release";
+
+        /// <summary>
+        /// Returns a user-friendly distribution name.
+        /// </summary>
+        internal static string? GetPrettyName(string filename = EtcOsReleasePath)
+        {
+            if (File.Exists(filename))
+            {
+                string[] lines;
+                try
+                {
+                    lines = File.ReadAllLines(filename);
+                }
+                catch
+                {
+                    return null;
+                }
+
+                // Parse the NAME, PRETTY_NAME, and VERSION fields.
+                // These fields are suitable for presentation to the user.
+                string? prettyName = null, name = null, version = null;
```

Comment:
Can you use `ReadOnlySpan<char>` here? We can save quite the allocations.

Reply:
I've changed it to use ROS.